### ISSUE TITLE
Record indices of survey responses?

### ISSUE THREAD

**Issue by goldpbear** [jodeleeuw/jsPsych#313]
Hi-- Thanks for your work on jsPsych. I've been using the framework to implement a survey recently, and it's made things very easy.

I have a question: I'm using the `jspsych-survey-multi-choice` plugin for our survey, and as far as I can tell, the plugin will only record survey responses using the full text of the chosen response. Is this the case?

If so, I think the option to record response indices rather than actual response text would be convenient. We're translating our survey into several languages, which means that recording raw response text in our database isn't practical. Plus, some of our response text is lengthy, so recording indices cuts down on storage size a bit.

I implemented a version of the plugin that records indices instead of response text. I'd be happy to clean it up and submit a PR if it's of interest.

Thanks!

**Comment by jodeleeuw** [comment]
There's a good chance that the various survey- plugins are going to be removed in the next version of jsPsych in favor of the new, much more flexible form plugin (#310). This is a feature that could be added to the form plugin without much trouble, right @GavinQ1? In fact, to make this as flexible as possible, we could add an option to record any data value instead of the label for radio/select/checkbox components. 

Something along the lines of:

```js
// records 1 if option1 selected, 2 if option2 selected.
"Question #4": {type: "radio", labels: ["option1", "option2"], correctAnswers: ["option1"], dataValues: [1,2] }

// records "foo" if option1, "bar" if option2
"Question #4": {type: "radio", labels: ["option1", "option2"], correctAnswers: ["option1"], dataValues: ["foo", "bar"] }
```

**Comment by goldpbear** [comment]
Ah, good to know--thanks. I agree that the ability to record arbitrary data values for survey responses would be a nice addition.

**Comment by GavinQ1** [comment]
@jodeleeuw  @goldpbear Great catch! In fact, this feature should have been here already. It's just that when I implemented the submitting form part, I mistakenly pushed 'product.label' instead of 'product.value' into the array (line 131).  I just fixed that and updated the example.

It would be something like this if you want arbitrary data values:
`
// records 1 if option1 selected, 2 if option2 selected.
"Question #4": {type: "radio", labels: ["option1", "option2"], correctAnswers: ["option1"], values: [1,2] }

// records "foo" if option1, "bar" if option2
"Question #4": {type: "radio", labels: ["option1", "option2"], correctAnswers: ["option1"], values: ["foo", "bar"] }
`
There is one thing should be noted though: 
if item.values is not assigned or item.values.length < item.label.length + item.image.length,
the item.values array will first take values from item.labels and then from item.images until item.values have the same size as item.labels + item.images
